src/quality-level-list.js

```diff
@@ -0,0 +1,135 @@
+import videojs from 'video.js';
+import document from 'global/document';
+
+/**
+ * A list of QualityLevels.
+ *
+ * interface QualityLevelList : EventTarget {
+ *   getter QualityLevel (unsigned long index);
+ *   readonly attribute unsigned long length;
+ *   readonly attribute long selectedIndex;
+ *
+ *   void addQualityLevel(QualityLevel qualityLevel)
+ *   void removeQualityLevel(QualityLevel remove)
+ *   QualityLevel? getQualityLevelById(DOMString id);
+ *
+ *   attribute EventHandler onchange;
+ *   attribute EventHandler onaddqualitylevel;
+ *   attribute EventHandler onremovequalitylevel;
+ * };
+ *
+ * @extends videojs.EventTarget
+ * @class QualityLevelList
+ */
+class QualityLevelList extends videojs.EventTarget {
+  constructor() {
+    super();
+
+    this.selectedIndex_ = -1;
+
+    let list = this; // eslint-disable-line
+
+    if (videojs.browser.IS_IE8) {
+      list = document.createElement('custom');
+      for (const prop in QualityLevelList.prototype) {
+        if (prop !== 'constructor') {
+          list[prop] = QualityLevelList.prototype[prop];
+        }
+      }
+    }
+
+    list.levels_ = [];
+  }
+
+  get selectedIndex() {
+    return this.selectedIndex_;
+  }
+
+  get length() {
+    return this.levels_.length;
+  }
+
+  addQualityLevel(qualityLevel) {
+    const index = this.levels_.length;
+
+    // Do not add duplicate quality levels
+    if (this.getQualityLevelById(qualityLevel.id)) {
+      return;
+    }
+
+    if (!('' + index in this)) {
+      Object.defineProperty(this, index, {
+        get() {
+          return this.levels_[index];
+        }
+      });
+    }
+
+    this.levels_.push(qualityLevel);
+
+    this.trigger({
+      qualityLevel,
+      type: 'addqualitylevel'
+    });
+  }
+
+  removeQualityLevel(remove) {
```

Comment:
Doesn't matter too much, but I usually find it easier to read (and consistent with other functions) if the param is named qualityLevel. Then the next var (currently named qualityLevel) could just be a boolean like didRemove (since we always have the reference to the removed qualityLevel from the param)